dask_cloudprovider/gcp/instances.py

```diff
@@ -293,8 +292,16 @@ async def start_scheduler(self):
         self.internal_ip, self.external_ip = await self.create_vm()
 
         if self.config.get("public_ingress", True):
-            # scheduler is publicly available
-            self.address = f"tcp://{self.external_ip}:8786"
+            if is_gce_env():
+                # if the client is running inside GCE environment
+                # it's better to use internal IP, which doesn't
+                # require firewall setup
+                self.address = f"tcp://{self.internal_ip}:8786"
```

Comment:
We have some duplication here. Can we combine this with the outer if statement.

Reply:
Good point, fixed.